### DISCUSSION TITLE
Using the autogenerator to set the username as the id field

### DISCUSSION THREAD

**Discussion by djakaria21** [Netflix/dgs-framework#1170] [Q&A]
Hi all,

I was trying to use mongodb with the dgs autogeneration tool but I wanted to set the id as the username instead of having an explicit id field.  I was wondering if anyone knew how to make the annotations for setting up the generated object with the document annotation and being able to set a String as the id field instead of explicitly having an id field in the schema.

**Comment by srinivasankavitha** [comment]
We don't support any generation of documentation annotation in the current tooling and framework.

**Reply by djakaria21** [reply]
So how do people use the tool when they're syncing it with mongodb or nosql servers?

**Comment by srinivasankavitha** [comment]
I'm assuming they generate their own data models when the mapping is different. We haven't invested in specifically supporting this scenario so far. What is the document annotation you are referring to?
